### ISSUE TITLE
aiokafka consumer throwing ValueError yet completing successfully

### ISSUE THREAD

**Issue by ZmeiGorynych** [aio-libs/aiokafka#184]
Hi,

when I try to do a fairly straightforward read of utf-8 -encoded json using aiokafka, I get the below error. Strangely, the program still completes successfully, also strangely, the error messages mention the producer when I think that part of the code should only be using the consumer. 

I'm using aiokafka 0.2.1 rather than 0.2.2 because of #166 

Is this related to #166 ?

Thanks a lot!
```
ERROR:asyncio:Task exception was never retrieved
future: <Task finished coro=<AIOKafkaProducer._send_produce_req() done, defined at C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\aiokafka\producer.py:331> exception=ValueError(<class 'struct.error'>,)>
Traceback (most recent call last):
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 17, in _unpack
    (value,) = unpack(f, data)
struct.error: unpack requires a bytes object of length 4

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\asyncio\tasks.py", line 239, in _step
    result = coro.send(None)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\aiokafka\producer.py", line 362, in _send_produce_req
    response = yield from self.client.send(node_id, request)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\aiokafka\client.py", line 361, in send
    request, expect_response=expect_response)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\aiokafka\conn.py", line 141, in send
    message = header.encode() + request.encode()
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\struct.py", line 34, in _encode_self
    [self.__dict__[name] for name in self.SCHEMA.names]
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 124, in encode
    for i, field in enumerate(self.fields)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 124, in <listcomp>
    for i, field in enumerate(self.fields)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 162, in encode
    [self.array_of.encode(item) for item in items]
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 162, in <listcomp>
    [self.array_of.encode(item) for item in items]
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 124, in encode
    for i, field in enumerate(self.fields)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 124, in <listcomp>
    for i, field in enumerate(self.fields)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 162, in encode
    [self.array_of.encode(item) for item in items]
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 162, in <listcomp>
    [self.array_of.encode(item) for item in items]
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 124, in encode
    for i, field in enumerate(self.fields)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 124, in <listcomp>
    for i, field in enumerate(self.fields)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\message.py", line 151, in encode
    size = Int32.decode(items)
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 50, in decode
    return _unpack('>i', data.read(4))
  File "C:\Users\Egor\Anaconda2\envs\py3k\lib\site-packages\kafka\protocol\types.py", line 20, in _unpack
    raise ValueError(error)
ValueError: <class 'struct.error'>
```

**Comment by tvoinarovskyi** [comment]
The same is reported in #176, but I fail to reproduce this error. I can reproduce #176 as a whole, but not this error... Can you put up a snippet? I do understand from you're words it's a simple one, but strangely I don't get this one...

**Comment by tvoinarovskyi** [comment]
Oh, wait. Could you try https://github.com/aio-libs/aiokafka/pull/180 this fix? This is probably the error, as it tries to encode from an incorrect position in the buffer.

**Comment by tvoinarovskyi** [comment]
Should be fixed in v0.2.3
